lib/mix/lib/mix/tasks/release.ex

```diff
@@ -1007,6 +1014,23 @@ defmodule Mix.Tasks.Release do
     release
   end
 
+  defp make_tar(release) do
+    tar_filename = "#{release.name}.tar.gz"
+    info(release, [:green, "* building tar ", :reset, tar_filename])
+    out_path = Path.join(release.path, tar_filename)
+
+    dirs =
+      ["bin", "lib", Path.join("releases", release.version), "erts-#{release.erts_version}"] ++
```

Comment:
You need to use `systools:make_tar` or duplicate its functionality because simply including `lib` in the tarball will include the libs used by other versions of the release that may have been built.